extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/input/table/SegmentsInputSliceTest.java

```diff
@@ -78,6 +78,31 @@ public void testSerde() throws Exception
     );
   }
 
+  @Test
+  public void testSerde2() throws Exception
+  {
+    final ObjectMapper mapper = TestHelper.makeJsonMapper()
+                                          .registerModules(new MSQIndexingModule().getJacksonModules());
+
+    final SegmentsInputSlice slice = new SegmentsInputSlice(
+        "myds",
+        ImmutableList.of(
+            new RichSegmentDescriptor(
+                Intervals.of("2000/P1M"),
+                Intervals.of("2000/P1M"),
+                "1",
+                0
+            )
+        ),
+        null
+    );
+
+    Assert.assertEquals(
+        slice,
+        mapper.readValue(mapper.writeValueAsString(slice), InputSlice.class)
```

Comment:
We would want to have a handCreated pojo without the key `  @JsonProperty("servedSegments"` rite ? because that would be sent over the write from the old controller.

Reply:
Changed the test case